CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeNetworkEngine.cs

```diff
@@ -334,6 +340,42 @@ private void ApplyNetworkEngineRequestConfiguration(UnityNativeRequest request)
             }
         }
 
+        private Dictionary<string, string> AddARP(Dictionary<string, string> headers)
+        {
+            string arpNamespaceKey = string.Format(UnityNativeConstants.Network.ARP_NAMESPACE_KEY,
+                _coreState.AccountInfo.AccountId, _coreState.DeviceInfo.DeviceId);
+            var arpJson = _preferenceManager.GetString(arpNamespaceKey, string.Empty);
+            if (string.IsNullOrEmpty(arpJson))
+                return headers;
+
+            Dictionary<string, object> arpDictionary = Json.Deserialize(arpJson) as Dictionary<string, object>;
+            if (arpDictionary == null || arpDictionary.Count == 0)
+                return headers;
+
+            var keysToRemove = new List<string>();
```

Comment:
Remove those keys before saving the ARP in preferences, to avoid doing it on every request.

Reply:
Key are getting removed below.

  foreach (var key in keysToRemove)
            {
                arpDictionary.Remove(key);
            }